examples/step-77/step-77.cc

```diff
@@ -554,7 +554,8 @@ namespace Step77
         // reach; but you might want to look into what other members the
         // SUNDIALS::KINSOL::AdditionalData class has and play with them).
         {
-          SUNDIALS::KINSOL<Vector<double>>::AdditionalData additional_data;
+          typename SUNDIALS::KINSOL<Vector<double>>::AdditionalData
+            additional_data;
```

Comment:
@peterrum  Why is this actually needed? Nothing here depends on a template parameters that is unknown at parse time. So `typename` is not necessary.